src/Compilers/CSharp/Test/Semantic/Semantics/ReadOnlyStructsTests.cs

```diff
@@ -1617,45 +1659,56 @@ public void ReadOnlyIndexer()
             var csharp = @"
 public struct S1
 {
-    public readonly int this[int i]
+    // ok
+    public readonly int this[int i] => i;
+}
+
+public struct S2
+{
+    // ok
+    public int this[int i]
     {
-        get => 42;
+        readonly get => i;
         set {}
     }
 }
-";
-            var comp = CreateCompilation(csharp);
-            comp.VerifyDiagnostics();
-        }
 
-        [Fact]
-        public void ReadOnlyExpressionIndexer()
-        {
-            var csharp = @"
-public struct S1
+public struct S3
 {
-    public readonly int this[int i] => 42;
+    // error
+    public int this[int i]
+    {
+        readonly get { return i; }
+        readonly set {}
+    }
 }
-";
-            var comp = CreateCompilation(csharp);
-            comp.VerifyDiagnostics();
-        }
 
-        [Fact]
-        public void ReadOnlyGetExpressionIndexer()
-        {
-            var csharp = @"
-public struct S1
+public struct S4
 {
-    public int this[int i]
+    // error
+    public readonly int this[int i]
     {
-        readonly get => 42;
-        readonly set {}
+        readonly get { return i; }
     }
 }
+
+public struct S5
+{
+    // error
+    public static readonly int this[int i] => i;
+}
 ";
             var comp = CreateCompilation(csharp);
-            comp.VerifyDiagnostics();
+            comp.VerifyDiagnostics(
+                // (21,16): error CS8659: Cannot specify 'readonly' modifiers on both accessors of property or indexer 'S3.this[int]'.
```

Comment:
Can we log a bug to add a quickfix to detect this and move them to the property/indexer outer part? Otherwise this just feels like the compiler is being obtuse.

Reply:
Seems reasonable. Whatever is currently done for accessibility modifiers is probably a good baseline since that’s what this behavior is based on.

Reply:
What's currently done for accessibility modifiers is that instead of a code fix which "consolidates" the duplicate accessibility modifiers we just add yet another accessibility modifier. So, it seems like we could do better on both..